packages/pipeline/src/Stage.tsx

```diff
@@ -0,0 +1,208 @@
+  const {
+    borderRadius,
+    chevron,
+    fontSize,
+    fontWeight,
+    gutter,
+    height,
+    lineHeight,
+  } = layout[size];
+
+  const { color, primary } = colors[variant];
+  const outerSize = height / 2;
+  const offset = outerSize + chevron.size * 2;
+
+  return cx(
+    getRootStyle({ size, variant }),
+    getChildStyle({ size, variant }),
+    css`
+      background-color: ${primary.backgroundColor};
+      color: ${color};
+      padding: ${gutter.vertical}px ${gutter.horizontal}px;
+      margin-right: ${offset}px;
+      font-size: ${fontSize}px;
+      font-weight: ${fontWeight};
+      line-height: ${lineHeight};
+      flex: 1 1 auto;
+      white-space: nowrap;
+      z-index: 1;
+
+      &:first-of-type {
+        border-top-left-radius: ${borderRadius}px;
+        border-bottom-left-radius: ${borderRadius}px;
+      }
+
+      &[data-stage-visible='false'] {
+        counter-increment: hiddenCount;
+      }
+    `,
+  );
+};
+
+const getStageChevronStyle = ({ size, variant }: StateForStyles): string => {
+  const { chevron, height } = layout[size];
+  const { primary } = colors[variant];
+  const outerSize = height / 2;
+
+  return cx(
+    getChevronStyle({ size, variant }),
+    css`
+      &::before {
+        background-color: ${primary.backgroundColor};
+        box-shadow: 0 0 0 ${chevron.size}px ${uiColors.white},
+          0 0 0 ${outerSize}px ${primary.backgroundColor};
+      }
+    `,
+  );
+};
+
+const getStatefulStyles = (state: StateForStyles) => ({
+  base: getBaseStyle(state),
+  chevron: getStageChevronStyle(state),
+});
+
+/**
+ * # Stage
+ *
+ * React Component to render an individual MongoDB Aggregation stage.
+ * To be used in conjunction with Pipeline component as a child.
+ *
+ * ```
+ * <Pipeline>
+ *   <Stage>$match</Stage>
+ *   <Stage>$addFields</Stage>
+ *   <Stage>$limit</Stage>
+ * </Pipeline>
+ * ```
+ * @param props.children Content that will appear inside of the Stage component.
+ * @param props.className Classname applied to Stage content container.
+ * @param props.intersectionNode The DOM node to use as the root node for the intersectionObserver. Defaults to window when null or undefined.
+ * @param props.size Alters the rendered size of the component.
+ * @param props.threshold Either a single number or an array of numbers which indicate at what percentage of the target's visibility the observer's callback should be executed.
+ * @param props.variant Alters the visual apperance of the component.
+ */
+const Stage = forwardRef(
+  (
+    {
+      children,
+      className,
+      intersectionNode,
+      size,
+      threshold,
+      variant,
+      ...rest
+    }: StageProps,
+    ref: Ref<HTMLLIElement>,
+  ): ReactElement => {
+    // Effects
+    const [setRef, isVisible] = useInView({
+      threshold,
+      root: intersectionNode,
+    });
+
+    const styles = getStatefulStyles({
+      size: size || Size.XSmall,
```

Comment:
Rather than setting default values here, we should be setting default values in the function parameters.

```js
({size = Size.XSmall}) => {...}
```

Reply:
Resolved by b5f3c0f